tree/ntuple/v7/src/RField.cxx

```diff
@@ -1387,10 +1388,49 @@ const ROOT::Experimental::RFieldBase::RColumnRepresentations &
 ROOT::Experimental::RField<float>::GetColumnRepresentations() const
 {
    static RColumnRepresentations representations(
-      {{EColumnType::kSplitReal32}, {EColumnType::kReal32}, {EColumnType::kReal16}}, {});
+      {{EColumnType::kSplitReal32}, {EColumnType::kReal32}, {EColumnType::kReal16}, {EColumnType::kReal32Trunc}}, {});
    return representations;
 }
 
+void ROOT::Experimental::RField<float>::GenerateColumns()
+{
+   const auto r = GetColumnRepresentatives();
+   const auto n = r.size();
+   fAvailableColumns.reserve(n);
+   for (std::uint16_t i = 0; i < n; ++i) {
+      auto &column = fAvailableColumns.emplace_back(Internal::RColumn::Create<float>(r[i][0], 0, i));
+      if (r[i][0] == EColumnType::kReal32Trunc) {
+         column->SetBitsOnStorage(fBitWidth);
+      }
+   }
+   fPrincipalColumn = fAvailableColumns[0].get();
+}
+
+void ROOT::Experimental::RField<float>::GenerateColumns(const RNTupleDescriptor &desc)
+{
+   std::uint16_t representationIndex = 0;
+   do {
+      const auto &onDiskTypes = EnsureCompatibleColumnTypes(desc, representationIndex);
+      if (onDiskTypes.empty())
+         break;
+
+      auto &column =
+         fAvailableColumns.emplace_back(Internal::RColumn::Create<float>(onDiskTypes[0], 0, representationIndex));
+      if (onDiskTypes[0] == EColumnType::kReal32Trunc) {
+         const auto &fdesc = desc.GetFieldDescriptor(GetOnDiskId());
+         const auto &coldesc = desc.GetColumnDescriptor(fdesc.GetLogicalColumnIds()[0]);
+         column->SetBitsOnStorage(coldesc.GetBitsOnStorage());
+      }
+      fColumnRepresentatives.emplace_back(onDiskTypes);
+      if (representationIndex > 0) {
+         fAvailableColumns[0]->MergeTeams(*fAvailableColumns[representationIndex]);
+      }
+
+      representationIndex++;
+   } while (true);
+   fPrincipalColumn = fAvailableColumns[0].get();
```

Comment:
We need to check for the case that (by accident) the field has no attached column at all (throw exception).

Reply:
Please disregard, already checked by `EnsureCompatibleColumnTypes`